.eslintrc.js

```diff
@@ -17,25 +17,29 @@ module.exports = {
   rules: {
     'no-console': 'error',
     'prettier/prettier': [
-      'error', {
-        'endOfLine': 'auto'
-      }
+      'error',
+      {
+        endOfLine: 'auto',
+      },
     ],
-    // TODO remove when done killing `any`s and making tsc strict
-    '@typescript-eslint/ban-ts-comment': 'off',
-    '@typescript-eslint/no-explicit-any': 'off',
-    '@typescript-eslint/no-unsafe-assignment': 'off',
-    '@typescript-eslint/no-unsafe-call': 'off',
-    '@typescript-eslint/no-unsafe-member-access': 'off',
-    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/explicit-function-return-type': 'warn',
+    '@typescript-eslint/no-confusing-non-null-assertion': 'warn',
+    '@typescript-eslint/no-explicit-any': 'warn',
+    '@typescript-eslint/no-extraneous-class': 'warn',
+    '@typescript-eslint/no-implicit-any-catch': 'warn',
+    '@typescript-eslint/no-invalid-void-type': 'warn',
+    '@typescript-eslint/prefer-ts-expect-error': 'warn',
+    '@typescript-eslint/type-annotation-spacing': 'warn',
+    '@typescript-eslint/typedef': 'warn',
+    '@typescript-eslint/unified-signatures': 'warn',
```

Comment:
Not a fan of using `warn`. I like linter rules that are `error`, or `off`:
  - Either we consider the rule valuable and we make it `error`, to have it breaking CI.
  - Or it's not always valuable / we're wishy-washy about it. At this point, making it `warn` only means it won't break CI, and then we're entering [broken window theory](https://en.wikipedia.org/wiki/Broken_windows_theory) territory: contributors not caring (or without a properly configured editor) won't see the warnings, will cause new ones, and then someday you come afterwards in a file full of them and you're annoyed because the file has become a warnings Christmas tree 😄.

Waddayathink? Is it reasonable to make all these error?

Reply:
It certainly is. I fixed all warnings, so it would be no impact. Just wasn't sure if we wanted to ease other devs into it. :)

Reply:
Go go `error` then. Other devs will fix linter errors too, then, and we can always remove rules that they/we find too much after some use.

Reply:
Done